### PR TITLE
FilenameBear: Add ignore_uppercase_filenames

### PR DESCRIPTION
**Pull request by sils1297**

### CODE REVIEW

File: bears/general/FilenameBear.py [hunk 1]
```diff
@@ -26,6 +29,9 @@ def run(self, filename, file, file_naming_convention: str="snake"):
             - ``camel`` (``thisIsCamelCase``)
             - ``pascal`` (``ThisIsPascalCase``)
             - ``snake`` (``this_is_snake_case``)
+        :param ignore_uppercase_filenames:
+            Wether or not to ignore fully uppercase filenames completely, e.g.
```
Comment (Adrianzatreanu) [inline bears/general/FilenameBear.py]: Whether*

File: bears/general/FilenameBear.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 from coalib.results.Result import Result
 from coalib.bearlib.naming_conventions import (
     to_camelcase, to_pascalcase, to_snakecase)
+from coalib.settings.Setting import typed_list
```
Comment (hypothesist) [inline bears/general/FilenameBear.py]: Unrelated change>

Comment (Adrianzatreanu) [inline bears/general/FilenameBear.py]: yep, not using this anywhere

**Comment by hypothesist** [comment]
https://github.com/coala-analyzer/coala-bears/pull/687/commits/52903a3a852ddf82ea4467bf4be9789a74a1da6f needs work

File: bears/general/FilenameBear.py (hunk 2 above)
Comment (sils1297) [inline bears/general/FilenameBear.py]: oh right I removed that change again...

**Comment by hypothesist** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/687/commits/255a126e4f388df393fbac7b7fe9b4711d57de58

**Comment by hypothesist** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/687/commits/8a1800fdb57b3253ff23eb4c091a5252bcc07768

**Comment by hypothesist** [comment]
@rultor merge
